src/api/app/components/notification_action_description_component.rb

```diff
@@ -54,4 +50,10 @@ def bs_request
                       @notification.notifiable.commentable
                     end
   end
+
+  def event_decision_description
+    decision_text = @notification.event_type.match(/Event::(\w+?)edDecision/)[1].downcase
```

Comment:
:thought_balloon: I find this code harder to follow than the previous one, it was dead easy to understand. In order of reducing the number of lines of that method, I think the right change is to get rid of the event case (I'm not saying I know how to do it, though...).

Reply:
OOP to the rescue. Each event is an object, so let's have a method on the event which is generating this notification action description. The downside would be to put some view logic in the models. I have to admit that I don't have a problem with this huge `case` statement, since it's quite simple to follow what happens.

Reply:
We had OOP in mind, but wanted to give the regular expression a try.

Changed to OOP. Please take a look again.

Reply:
On the first approach tests failed with:

```
  1) NotificationActionDescriptionComponent when the notification is for an Event::FavoredDecision renders the information about the favored decision
     Failure/Error: raise "Unexpected payload_keys :#{na.sort.join(', :')} (#{attribs.inspect}) provided during '#{self.class.name}' event creation. "
     
     RuntimeError:
       Unexpected payload_keys :fake ({"fake"=>"payload"}) provided during 'Event::FavoredDecision' event creation. 
     # ./app/models/event/base.rb:138:in `check_left_attribs'
     # ./app/models/event/base.rb:152:in `set_payload'
     # ./app/models/event/base.rb:124:in `initialize'
     # ./app/models/notification.rb:23:in `event'
     # ./app/components/notification_action_description_component.rb:56:in `event_decision_description'
     # ./app/components/notification_action_description_component.rb:36:in `block in call'
     # ./app/components/notification_action_description_component.rb:17:in `call'
     # ./spec/components/notification_action_description_component_spec.rb:213:in `block (3 levels) in <top (required)>'
```

I found we could use `@notification.notifiable.kind`, and adapt it. Please take a look again.